code/modules/mob/living/status_procs.dm

```diff
@@ -423,6 +423,12 @@
 		blind_eyes(1)
 	ADD_TRAIT(src, TRAIT_BLIND, source)
 
+/mob/living/proc/cure_blind_echolocation(list/sources)
+	REMOVE_TRAIT(src, TRAIT_BLIND_ECHO, sources)
+
+/mob/living/proc/become_blind_echolocation(source)
```

Comment:
Why does this not call ``become_blind(source)``?

Reply:
i stated on discord, its necessary of echolocation to function and we want to be able to add the component without the trait, making a trait every single time we want to add echolocation using a component would be silly, its better to do it in echolocation and not in the trait so that it always works

Reply:
Can't this simply just be the trait then, and remove this proc entirely?

Reply:
wdym

Reply:
Instead of doing ``become_blind_echolocation()``, do ``ADD_TRAIT(src, TRAIT_BLIND_ECHO, insert source here)``, since it does not actually make you blind, it makes no sense to be one

Reply:
is it not just better form to make it inline with other trait procs? idk why we'd want to specifically have to do it in a weird way for echolocation just to save on four lines of code

Reply:
This is not a status proc, this is a trait that affects people when they become blind. Blind is a status proc